src/App.tsx

```diff
@@ -1,24 +1,210 @@
+        setError('');
+      }, 3000);
+    }
+  };
+
+  const onDeleteComplete = () => {
+    const FilterCompletedTodos = todos.filter((todo) => todo.completed);
+
+    FilterCompletedTodos.map(async (todo) => {
+      try {
+        await deleteTodo(todo.id);
+        setTodos(todos.filter((task) => !task.completed));
+      } catch {
+        setError('Unable to remove todo');
+        setTimeout(() => {
+          setError('');
+        }, 3000);
+      }
+    });
+  };
+
+  const toggleAllCompleted = () => {
+    const allCompleted = todos.every((todo) => todo.completed);
+
+    if (allCompleted) {
+      todos.map((todoEl) => {
+        return handleUpdate(todoEl.id, { completed: false });
+      });
+    } else {
+      const notCompleted = todos.filter((el) => !el.completed);
+
+      notCompleted.map((todoElement) => {
+        return handleUpdate(todoElement.id, { completed: true });
+      });
+    }
+  };
+
+  const handleSubmit = (e: { preventDefault: () => void }) => {
+    e.preventDefault();
+    if (!title || title.trim() === '') {
+      setError('Tittle can not be empty');
+      setTimeout(() => {
+        setError('');
+      }, 3000);
+
+      return;
+    }
+
+    addTodo(title);
+    setTitle('');
+  };
+
+  const changeHandler = (e: {
+    target: { value: React.SetStateAction<string> };
+  }) => {
+    setTitle(e.target.value);
+  };
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          onSubmitForm={handleSubmit}
+          title={title}
+          onChangeTitle={changeHandler}
+          onToggleAllCompleted={toggleAllCompleted}
+        />
+
+        {todos.length > 0 && (
+          <>
+            <section className="todoapp__main">
+              <TodoList
+                todos={currentTodos}
+                onDeleteTodo={removeTodo}
+                temporaryTodo={temporaryTodo}
+                onUpdateTodo={handleUpdate}
+              />
+            </section>
+
+            <footer className="todoapp__footer">
+              <span className="todo-count">{`${activeTodosСount}  items left`}</span>
+
+              <FilterTodo filter={filter} onFilterChange={setFilter} />
+
+              {completedTodos ? (
+                <button
+                  type="button"
+                  className="todoapp__clear-completed"
+                  onClick={onDeleteComplete}
+                >
+                  Clear completed
+                </button>
+              ) : (
+                ''
+              )}
```

Comment:
Why do you need a ternary operator here? 
```suggestion
              {completedTodos && (
                <button
                  type="button"
                  className="todoapp__clear-completed"
                  onClick={onDeleteComplete}
                >
                  Clear completed
                </button>
              )}
```